genesis/process/intermediate/deployLibrarySC.go

```diff
@@ -0,0 +1,149 @@
+// last 2 bytes
+func NewDeployLibrarySC(arg ArgDeployLibrarySC) (*deployLibrarySC, error) {
+	if check.IfNil(arg.Executor) {
+		return nil, genesis.ErrNilTxExecutionProcessor
+	}
+	if check.IfNil(arg.PubkeyConv) {
+		return nil, genesis.ErrNilPubkeyConverter
+	}
+	if check.IfNil(arg.BlockchainHook) {
+		return nil, process.ErrNilBlockChainHook
+	}
+	if check.IfNil(arg.ShardCoordinator) {
+		return nil, genesis.ErrNilShardCoordinator
+	}
+
+	base := &baseDeploy{
+		TxExecutionProcessor: arg.Executor,
+		pubkeyConv:           arg.PubkeyConv,
+		blockchainHook:       arg.BlockchainHook,
+		emptyAddress:         make([]byte, arg.PubkeyConv.Len()),
+		getScCodeAsHex:       getSCCodeAsHex,
+	}
+
+	dp := &deployLibrarySC{
+		baseDeploy:       base,
+		shardCoordinator: arg.ShardCoordinator,
+	}
+
+	return dp, nil
+}
+
+func GenerateInitialPublicKeys(
+	baseAddress []byte,
+	shardCoordinator sharding.Coordinator,
+	allShards bool,
+) [][]byte {
+	addressLen := len(baseAddress)
+
+	newAddresses := make([][]byte, 0)
+	i := uint8(0)
+	for ; i < core.MaxShardNumber-1; i++ {
+		shardInBytes := []byte{0, i}
+		tmpAddress := string(baseAddress[:(addressLen-core.ShardIdentiferLen)]) + string(shardInBytes)
+
+		newShardID := shardCoordinator.ComputeId([]byte(tmpAddress))
+		if !allShards && newShardID != shardCoordinator.SelfId() {
+			continue
+		}
+
+		newAddresses = append(newAddresses, []byte(tmpAddress))
+	}
+
+	shardInBytes := []byte{0, i}
+	tmpAddress := string(baseAddress[:(addressLen-core.ShardIdentiferLen)]) + string(shardInBytes)
+
+	newShardID := shardCoordinator.ComputeId([]byte(tmpAddress))
+	if allShards || newShardID == shardCoordinator.SelfId() {
+		newAddresses = append(newAddresses, []byte(tmpAddress))
+	}
+
+	return newAddresses
+}
+
+// Deploy will try to deploy the provided smart contract
+func (dp *deployLibrarySC) Deploy(sc genesis.InitialSmartContractHandler) error {
+	code, err := dp.getScCodeAsHex(sc.GetFilename())
+	if err != nil {
+		return err
+	}
+
+	resultingScAddresses := make([][]byte, 0)
+	newAddresses := GenerateInitialPublicKeys(genesis.InitialDNSAddress, dp.shardCoordinator, false)
+	for _, newAddress := range newAddresses {
+		scAddress, err := dp.deployForOneAddress(sc, newAddress, code, sc.GetInitParameters())
+		if err != nil {
+			return err
+		}
+
+		resultingScAddresses = append(resultingScAddresses, scAddress)
+		err = dp.changeOwnerAddress(scAddress, newAddress, sc.OwnerBytes())
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (dp *deployLibrarySC) changeOwnerAddress(
+	scAddress []byte,
+	currentOwner []byte,
+	newOwner []byte,
+) error {
+	nonce, err := dp.GetNonce(currentOwner)
+	if err != nil {
+		return err
+	}
+
+	txData := []byte(core.BuiltInFunctionChangeOwnerAddress + "@" + hex.EncodeToString(newOwner))
+	err = dp.ExecuteTransaction(nonce, currentOwner, scAddress, big.NewInt(0), txData)
+	if err != nil {
+		return err
+	}
+
+	account, ok := dp.AccountExists(scAddress)
```

Comment:
Not sure why not checking this before `ExecuteTransaction` instead.

Reply:
copy paste verification remained.

Reply:
it is checked before. check is needed here as well as you do not want the node to panic. although that means super big coding problem.